File: security/src/main/java/ru/virgil/security/InternalSecurityToken.java
package ru.virgil.security;

import org.springframework.security.authentication.AbstractAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;
import ru.virgil.security.entity.SecurityUser;

import javax.annotation.Nullable;
import java.util.Collection;

public class InternalSecurityToken extends AbstractAuthenticationToken {

    private final SecurityUser securityUser;

    /**
     * Авторизованный внутренний токен, который уходит вглубь системы, не должен знать авторизационных данных.
     * <p>
     * Хранит в себе ID и сам объект пользователя
     */
    public InternalSecurityToken(Collection<? extends GrantedAuthority> authorities, SecurityUser securityUser) {
        super(authorities);
        this.securityUser = securityUser;
        setAuthenticated(true);
    }

    /**
     * @return null, т.к. внутри сервера мы не должны знать никакой авторизационной информации пользователя
     */
    @Override
    @Nullable
    public Void getCredentials() {
        return null;
    }

    @Override
    public SecurityUser getPrincipal() {
        return securityUser;
    }
}

File: security/src/main/java/ru/virgil/security/InternalSecurityToken.kt
package ru.virgil.security

import org.springframework.security.authentication.AbstractAuthenticationToken
import org.springframework.security.core.GrantedAuthority
import ru.virgil.security.entity.SecurityUser

class InternalSecurityToken(
    authorities: Collection<GrantedAuthority>,
    private val securityUser: SecurityUser,
) : AbstractAuthenticationToken(authorities) {

    /**
     * Авторизованный внутренний токен, который уходит вглубь системы, не должен знать авторизационных данных.
     *
     *
     * Хранит в себе ID и сам объект пользователя
     */
    init {
        isAuthenticated = true
    }

    /**
     * @return null, т.к. внутри сервера мы не должны знать никакой авторизационной информации пользователя
     */
    override fun getCredentials(): Void? = null

    override fun getPrincipal(): SecurityUser = securityUser
}
